woocommerce/woocommerce
# [PHP 8.1] Fix code that throws deprecation notices in tests in PHP 8.1

### All Submissions:


### Changes proposed in this Pull Request:

Do the necessary code fixes so that the WooCommerce unit tests don't throw deprecation notices in PHP 8.1. The deprecation notices fixed are:

- Various "passing null to parameter (...) is deprecated".
- Usage of `strftime`.
- Using `false` as if it was an empty array.
- Mismatching return type of implemented interfaces, that's fixed by adding `#[\ReturnTypeWillChange]` attributes.

Note that this pull request alone is not enough to fix all the tests:

1. Blocks and Action Scheduler will need fixes too (pull request links in the testing instructions).
2. WordPress core itself isn't currently compatible with PHP 8.1. Some tests are skipped temporarily (see `skip_on_php_8_1` method introduced in `class-wc-unit-test-case.php`), only in PHP 8.1, until that's no longer the case.

Once Blocks and Action Scheduler have got the required fixes we could extend the GitHub action that run the tests to include PHP 8.1, keeping the WordPress-offending ones disabled for the time being.

Closes: https://github.com/woocommerce/woocommerce/issues/31270
Closes: https://github.com/woocommerce/woocommerce/issues/31386

**NOTE:** A lot of code formatting changes were needed for the code sniffer to be happy. These are in a separate commit for easier review.

### How to test the changes in this Pull Request:

1. Ensure that the fixes required for Blocks and Action Scheduler are applied ([PR for blocks](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5287) - [PR for Action Scheduler](https://github.com/woocommerce/action-scheduler/pull/791)).
2. Run the unit tests in PHP 7.
3. Perform [the changes needed to run tests in PHP 8](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests#running-tests-in-php-8).
4. Run the unit tests in both PHP 8.0 and PHP 8.1.

### Other information:


### Changelog entry

> Fix - Code that produces deprecation notices when running unit tests in PHP 8.1.

### FOR PR REVIEWER ONLY:
